https://bugs.gentoo.org/875122

First commit: https://github.com/apache/tomcat-connectors/commit/e719874b21a526cb3447c747b9901719cf46eafe
The rest: https://github.com/apache/tomcat-connectors/pull/6

From: Mark Thomas <markt@apache.org>
Date: Wed, 30 Jun 2021 10:58:58 +0100
Subject: [PATCH] Fix BZ 64878 - Fix build in MaxOS

With thanks to don@isle.plus at
https://bz.apache.org/bugzilla/show_bug.cgi?id=64753#c13

https://bz.apache.org/bugzilla/show_bug.cgi?id=64878
--- a/scripts/build/jk_common.m4
+++ b/scripts/build/jk_common.m4
@@ -35,6 +35,7 @@ changequote([, ])dnl
 AC_MSG_CHECKING(size of $2)
 AC_CACHE_VAL(AC_CV_NAME,
 [AC_TRY_RUN([#include <stdio.h>
+#include <stdlib.h>
 $1
 main()
 {

From f5403f13b52c50a19afa4e16dbbb66c31dce565b Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Mon, 17 Oct 2022 03:46:06 +0100
Subject: [PATCH 1/2] Fix -Wimplicit-int

Becomes fatal with Clang 15.
--- a/scripts/build/jk_common.m4
+++ b/scripts/build/jk_common.m4
@@ -37,7 +37,7 @@ AC_CACHE_VAL(AC_CV_NAME,
 [AC_TRY_RUN([#include <stdio.h>
 #include <stdlib.h>
 $1
-main()
+int main()
 {
   FILE *f=fopen("conftestval","w");
   if (!f) exit(1);
From c896afdb44d26186bd93301827f4b0b7888ac116 Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Mon, 17 Oct 2022 03:51:06 +0100
Subject: [PATCH 2/2] configure.ac: Fix more implicit function declarations

Becomes fatal in Clang 16.

Signed-off-by: Sam James <sam@gentoo.org>
--- a/configure.ac
+++ b/configure.ac
@@ -303,9 +303,11 @@ AC_CHECK_HEADERS(sys/filio.h)
 AC_DEFUN([JK_CHECK_SETSOCKOPT], [
 AC_MSG_CHECKING(whether to use $1 with setsockopt())
 AC_TRY_RUN([
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/time.h>
+#include <stdlib.h>
 
 int main(void)
 {
@@ -344,6 +346,7 @@ AC_TRY_RUN([
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/time.h>
+#include <stdlib.h>
 
 int main(void)
 {
@@ -395,6 +398,7 @@ AC_TRY_RUN([
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/time.h>
+#include <stdlib.h>
 
 int main(void)
 {
@@ -416,6 +420,7 @@ AC_TRY_RUN([
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
+#include <stdlib.h>
 
 int main(void)
 {
